Correct filtering of OpenStack LBaaS resources

FengyunPan opened the pull request:
Neutron's API ignores unknown paramaters. When listing pools etc,
K8S attempts to filter on "LoadBalancerID" which is not a valid
filter. As such, it is ignored by Neutron, and a list of all
pools is returned. Then K8S proceeds all the pools. For example,
EnsureLoadBalancerDeleted() delete all the pools and members.
Instead, we sholud double check the resources really belong to
the Neutron LB.

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md and developer guide https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md
2. If you want *faster* PR reviews, read how: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/faster_reviews.md
3. Follow the instructions for writing a release note: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes
-->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes #43055 

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->
```release-note
```

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/43056)
<!-- Reviewable:end -->

FengyunPan commented:
cc @Kiall Mac Innes  <kiall@hpe.com>  
cc @kubernetes/kubernetes-sig-openstack
Hello, I'd like you to do a code review, thank you.

kiall commented:
I'm about to hop on a plane, wont be back for 2 weeks.. I've asked a colleague (@grahamhayes) to take a look and test it..

grahamhayes commented:
is this different to #38959 ?

xsgordon commented:
My initial reaction was the same as that of @grahamhayes though I know there have been multiple variations of this behavior on different paths through the code. @FengyunPan can you take a look at https://github.com/kubernetes/kubernetes/pull/38959 and see if addresses your issue?

/cc @k8s-sig-openstack-bugs 

@idvoretskyi can you add the sig/openstack label ?

FengyunPan commented:
Hi @grahamhayes  @xsgordon ,  thank for your review, that PR  #38959 may address my issue, but my PR update UpdateLoadBalancer() too. I think the best way is to use a uniform function to get OpenStack LB resources.

FengyunPan commented:
/release-note-none

dims commented:
@k8s-bot ok to test

FengyunPan commented:
@k8s-bot bazel test this

Commit:
Corect filtering of OpenStack LBaaS resources

Neutron's API ignores unknown paramaters. When listing pools etc,
K8s attempts to filter on "LoadBalancerID" which is not a valid
filter. As such, it is ignored by Neutron, and a list of all
pools is returned. Then K8s proceeds all the pools. For example,
EnsureLoadBalancerDeleted() delete all the pools and members.
Instead, we sholud double check the resources really belong to
the Neutron LB.

FengyunPan commented:
@k8s-bot bazel test this
